specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,176 @@
+# Background
+
+To improve the developer experience for customizing non-client regions, WebView2 is 
+working to support using DOM elements as non-client regions. We currently have limited 
+support for custom title bars aka draggable regions and are working on building out support 
+for caption controls and resize regions. 
+
+For security and flexibility, we want developers to be able to enable or disable all 
+custom non-client functionality per WebView. Non-client functionality will affect that 
+app window’s size and position so it’s important that developers can definitively 
+toggle access. This can be achieved in a limited way using a feature flag, but feature 
+flags are applied per WebView2 environment, thus, an API on the WebView2 to enable/disable 
+non-client support via a setting is the better solution.
+
+# Description
+`IsNonClientRegionSupportEnabled` defaults to `FALSE`. Disabling/Enabling 
+`IsNonClientRegionSupportEnabled` takes effect after the next navigation. Currently, draggable 
+regions is the only non-client experience we have implemented. Eventually, this setting will 
+expand to enable other non-client functionality, such as resize and caption controls. 
```

Comment:
Is there any possibility that somebody might want `app-region: drag` integration but not other future non-client region support?

If so, then maybe an enum or flags so you write
```csharp
webview2.NonClientRegionSupport = CoreWebView2NonClientRegionSupport.None;
webview2.NonClientRegionSupport = CoreWebView2NonClientRegionSupport.TitleBar;
webview2.NonClientRegionSupport = CoreWebView2NonClientRegionSupport.TitleBar | CoreWebView2NonClientRegionSupport.Resizing;
```

But maybe the answer is "No, there's no need to have fine-grained control. If you enable non-client regions, then you have to accept *all* possible future non-client regions. In practice, you're doing this only with sites that you already trust and probably control, so your mitigation is simply "Don't use those features in your HTML."

Reply:
I believe your second paragraph is correct. There may be finer grained control in the CSS but we don't need finer grained control over allowing different aspects of the feature. @tochukwuIbeEkeocha please follow up with your PM

Reply:
I'm interpreting your question in two ways, I'm not sure which you are asking so I'll answer both. 

There is finer grain control through CSS. For example, you can have multiple HTML elements one can be marked as `app-region: drag`, another `app-region: minimize`, then another `app-region: close`, etc... In this way, if an app developer does not wish to support one of these possible non-client regions, they can exclude it from their app.

On the other hand, it is not (currently) possible through Weview2 to have us ignore the presence of a non-client region specified with `app-region` on an HTML element.

Reply:
> So really the app-region value should be named something like title-bar (`drag` does a lot more than drag), but that's an existing name not something we're introducing?

Yes, the value name drag already exists, It is not being introduced by this change